smk/mags_generation.smk

```diff
@@ -681,20 +689,31 @@ rule phylophlan_taxonomy_for_genome_collection:
         """
         time (
             phylophlan_assign_sgbs -i {input.genomes} --nproc {threads} -d {wildcards.db_version} -o taxonomy --database_folder {params.db_folder}
+            echo -e "taxa_ID\tkingdom\tphylum\tclass\torder\tfamily\tgenus\tspecies" > taxonomy.tsv.fixed
+            cut -f1,2 taxonomy.tsv \
+                    | tail -n +2 \
+                    | perl -lane '($id, undef, $tax, $dist) = split(/:/, $F[1]); @tax = map {{s/.__//; $_}} split(/\|/, $tax); pop(@tax); $, = "\t"; print($dist, $F[0], @tax);' \
+                    | perl -lane '$dist = shift(@F); $F[7]="NA" if $dist>0.05; $F[6]="NA" if $dist>0.1; $F[5]="NA" if $dist>0.2; print join("\t", @F)' \
+                    >> taxonomy.tsv.fixed
         ) >& {log}
-        rsync -a taxonomy.tsv {output}
+        rsync -a taxonomy.tsv {output.orig}
+        rsync -a taxonomy.tsv.fixed {output.fixed}
         """
```

Comment:
The perl lines produce empty lines for the unknown taxa, as phylophlan outputs non-tab separated lines in the end of the output tsv for the MAGs, where no corresponding reference is found:
`aaez.MetaSPAdes.k21-99.S1_4057 is close to an SGB not present in the SGB.Jun23 database
aaez.MEGAHIT.meta-large.S1_4484 is not close to any SGB present in the SGB.Jun23 database`

Reply:
This is outdated. I have added a patch to fix this already. And this part has moved to a different smk file too!